docs/guide/nts.md

```diff
@@ -0,0 +1,109 @@
+
+Next, a root certificate needs to be generated for this key:
+```sh
+openssl req -x509 -new -nodes -key myCA.key -sha256 -days 1825 -out myCA.pem
+```
+This will ask for a bunch of information. NTPD-rs does not impose particular requirements to the information in your root certificate, so feel free to leave things empty. The above command gives the certificate a period validity of 5 years. If a different lifetime is desired, change the number following the `-days` argument to the number of days the certificate should be valid. Note that this command bases validity on the current system time, so ensure it is reasonably accurate.
+
+**server certificate**: Having set up our CA, we can now generate the server key and certificate. First, we generate a private key for the server:
+```sh
+openssl genrsa -out ntpd-rs.test.key 2048
+```
+As this key will be used by the server, we can't protect it with a password, so the above command doesn't enable password protection for it.
+
+Next, just like when requesting a certificate from an external CA, we need to create a certificate signing request.
+```sh
+openssl req -new -key ntpd-rs.test.key -out ntpd-rs.test.csr
+```
+Again, NTPD-rs does not impose requirements on the information requested here, so feel free to leave things empty.
+
+Next, we will need to create a file that specifies what our CA thinks the key should be usable for. Create a text file called `ntpd-rs.ext` with the following contents:
+```ini
+authorityKeyIdentifier=keyid,issuer
+basicConstraints=CA:FALSE
+keyUsage = keyEncipherment
+subjectAltName = @alt_names
+
+[alt_names]
+DNS.1 = ntpd-rs.test
+```
+You can change ntpd-rs.test to the domain name of your server.
+
+Finally, we generate the certificate with
+```sh
+openssl x509 -req -in ntpd-rs.test.csr -CA myCA.pem -CAkey myCA.key -CAcreateserial -out ntpd-rs.test.crt -days 1825 -sha256 -extfile ntpd-rs.ext
+```
+The above command gives the certificate a period validity of 5 years. If a different lifetime is desired, change the number following the `-days` argument to the number of days the certificate should be valid.
+
+Finally, the ntpd-rs server requires the full certificate chain to be provided to it, so let's generate it with
+```sh
+cat ntpd-rs.test.crt myCA.pem > ntpd-rs.test.chain
+```
+
+### Configuring key exchange
+
+We are now ready to configure the key exchange part of NTS. This is used by clients when setting up a connection.
+```toml
+[[nts-ke-server]]
+key-exchange-listen = "[::]:4460"
+certificate-chain-path = "ntpd-rs.test.chain"
+private-key-path = "ntpd-rs.test.key"
+```
+Note that this needs to listen on its own port. Furthermore, the key exchange happens over a TCP connection rather than over UDP packets.
+
+When configured as above, it is important to ensure that the TLS private key can only be read by NTP-daemon and trusted system administrators. An attacker can use this key material to compromise all connections to clients.
```

Comment:
- "TLS" has not been introduced at this point
- NTP daemon does not require the `-`